docs/versioned_docs/version-zenith/quickstart/842901-test.mdx

```diff
@@ -0,0 +1,56 @@
+---
+slug: /quickstart/842901/test
+hide_table_of_contents: true
+title: "Add a function to test the application"
+---
+
+import PartialExperimentalDocs from '../partials/_experimental.mdx';
+
+# Quickstart
+
+<PartialExperimentalDocs />
+
+## Add a function to test the application
+
+The function in the previous example was illustrative - you wouldn't ever use it in the real world! So, the next step is to write a Dagger function to do something more useful, like running an application's tests.
+
+Add a new `Test()` function to `main.go`, which runs the application's unit test suite:
+
+This function does the following:
+
+- It uses the `dag` client's `Container().From()` method to initialize a new container from a base image. In this example, the base image is again `node:18-slim`. This method returns a `Container` initialized from that base image.
+- When a module is loaded into Dagger, it includes the directory tree at which `dagger.json` is located. In this example, since `dagger.json` is co-located with the application source code directory, that directory is auto-included in the module container's filesystem and can be accessed via `dag.Host().Directory()`.
+    - The function code uses the `Container.WithDirectory()` method to mount the source code directory from the module container's filesystem at the `/src` mount point in the container, and the `Container.WithWorkdir()` method to set the working directory to that mount point.
+    - The `dag.Host().Directory()` method queries the module container's environment and returns a reference to the application source code directory in the module container's filesystem after excluding the `.git` and `node_modules` sub-directories.
+- It uses the `Container.WithExec()` method to define the commands to install dependencies and run tests in the container - in this case, the commands `npm install` and `npm run test:unit run`.
+- It uses the `Container.Stdout()` method to return the output of the last executed command (the test results). If tests pass, the output shows the list of passed tests. If not, a non-nil `error` is returned, which propagates to the Dagger CLI and lets it know that one or more tests failed.
+
+:::note
+It is important to note that the `dag.Host().Directory()` method does not read from the CLI caller's filesystem, but from the module container's filesystem.
+:::
```

Comment:
Can remove now since the recommended way to read files from the module's source is the `dag.CurrentModule().Source()` api

Reply:
Removed